apps/dep-graph-client/src/app/graph.ts

```diff
@@ -0,0 +1,384 @@
+            'border-width': '1px',
+            'text-halign': 'center',
+            'text-valign': 'center',
+            'padding-left': '16px',
+          },
+        },
+        {
+          selector: 'node.focused',
+          style: {
+            color: NrwlPallette.twilight,
+            'border-color': NrwlPallette.twilight,
+          },
+        },
+        {
+          selector: 'node.affected',
+          style: {
+            'border-color': NrwlPallette.red,
+          },
+        },
+        {
+          selector: 'edge',
+          style: {
+            width: '1px',
+            'line-color': NrwlPallette.black,
+            'curve-style': 'straight',
+            'target-arrow-shape': 'triangle',
+            'target-arrow-fill': 'filled',
+            'target-arrow-color': NrwlPallette.black,
+          },
+        },
+        {
+          selector: 'edge.affected',
+          style: {
+            'line-color': NrwlPallette.red,
+            'target-arrow-color': NrwlPallette.red,
+          },
+        },
+        {
+          selector: 'edge.implicit',
+          style: {
+            label: 'implicit',
+            'font-size': '8px',
+            'edge-text-rotation': 'autorotate',
+          },
+        },
+        {
+          selector: 'edge.dynamic',
+          style: {
+            'line-dash-pattern': [5, 5],
+            'line-style': 'dashed',
+          },
+        },
+        {
+          selector: ':parent',
+          style: {
+            'background-opacity': 0.5,
+            'background-color': NrwlPallette.twilight,
+            'border-color': NrwlPallette.black,
+            label: 'data(label)',
+            'text-halign': 'center',
+            'text-valign': 'top',
+            'font-weight': 'bold',
+            'font-size': '24px',
+          },
+        },
+      ],
+    });
+
+    this.graph.anywherePanning();
+
+    this.graph.on('zoom', () => {
+      if (this.openTooltip) {
+        this.openTooltip.hide();
+        this.openTooltip = null;
+      }
+    });
+  }
+
+  render() {
+    this.generateCytoscapeLayout();
+    this.addTooltips();
+  }
+
+  addTooltips() {
+    this.graph.$('node:childless').on('click', (event) => {
+      const node = event.target;
+
+      let ref = node.popperRef(); // used only for positioning
+
+      // unfortunately, a dummy element must be passed as tippy only accepts a dom element as the target
+      // https://github.com/atomiks/tippyjs/issues/661
+      let dummyDomEle = document.createElement('div');
+
+      let tip = tippy(dummyDomEle, {
+        trigger: 'manual', // call show() and hide() yourself
+        lazy: false, // needed for onCreate()
+        interactive: true,
+        appendTo: document.body,
+        onCreate: (instance) => {
+          instance.popperInstance.reference = ref;
+        }, // needed for `ref` positioning
+
+        content: () => {
+          let content = document.createElement('div');
+
+          content.innerHTML = `
```

Comment:
Hm... instead of `innerHTML` we could append dom...

```
const heading = createHeading(node);
function createHeading(node) {
  const ele = document.createElement('h4');
  ele.innerText = node.attr('type');
  return ele;
}
```

Also, if possible could we translate the click event and resolve to data in our own format... so that we don't have to use the DOM so much? something like...

```
on('click', (event) => {
  const project = this.projects[event.target.id];
  showTooltip(project, event.target);
})
```